deploy/test-environments/fleet_api/src/agents_enrolled.py

```diff
@@ -36,23 +48,57 @@ def get_actual_agents() -> dict:
     return policies_dict
 
 
-def verify_agents_enrolled() -> bool:
+def verify_agent_count(expected: dict, actual: dict) -> bool:
     """
     Verify that the expected number of agents are enrolled
     """
-    expected = get_expected_agents()
-    actual = get_actual_agents()
     result = True
-    for policy_id, expected_count in expected.items():
+    for policy_id, expected_agents in expected.items():
         if policy_id not in actual:
             result = False
             logger.info(f"Policy {policy_id} not found in the actual agents mapping")
-        elif actual[policy_id] != expected_count:
+        elif actual[policy_id] != expected_agents.count:
             result = False
-            logger.info(f"Policy {policy_id} expected {expected_count} agents, but got {actual[policy_id]}")
+            logger.info(f"Policy {policy_id} expected {expected_agents.count} agents, but got {actual[policy_id]}")
+        else:
+            logger.info(f"Policy {policy_id} has {actual[policy_id]} agents as expected")
     return result
 
 
+def verify_agent_tags(agent, expected_agents) -> bool:
+    """
+    Verify that the agent has the expected tags
+    """
+    expected_tags = []
+    if agent.policy_id in expected_agents:
+        expected_tags = expected_agents[agent.policy_id].tags
+    for pattern in expected_tags:
+        pattern_exist = False
+        for tag in agent.tags:
+            if re.match(pattern, tag):
+                pattern_exist = True
+                break
+        if not pattern_exist:
+            logger.warning(f"Agent {agent.id} does not have the expected tag {pattern}")
+            return False
```

Comment:
`any()` returns True if any of the tags in agent.tags match the current pattern.
```suggestion
    for pattern in expected_tags:
        if not any(re.match(pattern, tag) for tag in agent.tags):
            logger.warning(f"Agent {agent.id} does not have the expected tag {pattern}")
            return False
```